stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt

```diff
@@ -54,6 +58,16 @@ internal class PaymentOptionsViewModel(
         _processing.postValue(false)
     }
 
+    override fun onFatal(throwable: Throwable) {
+        _fatal.value = throwable
+        _paymentOptionResult.value = PaymentOptionResult.Failed(throwable)
+    }
+
+    override fun onUserCancel() {
+        _paymentOptionResult.value =
+            PaymentOptionResult.Canceled(mostRecentError = _fatal.value)
```

Comment:
We don't read the value of `mostRecentError` anywhere and it's only set here. I think we can remove that from `PaymentOptionResult.Canceled`, similarly to how we did to `PaymentSheetResult.Canceled`. Then we can also get rid of `_fatal` which is only read here.

Reply:
I will make this a separate issue.